packages/build-tools/src/builders/android.ts

```diff
@@ -66,6 +69,13 @@ async function buildAsync(ctx: BuildContext<Android.Job>): Promise<void> {
     await runHookIfPresent(ctx, Hook.POST_INSTALL);
   });
 
+  const resolvedExpoUpdatesRuntimeVersion = await ctx.runBuildPhase(
+    BuildPhase.CALCULATE_EXPO_UPDATES_RUNTIME_VERSION,
+    async () => {
+      return await resolveRuntimeVersionForExpoUpdatesIfConfiguredAsync(ctx);
+    }
+  );
```

Comment:
The runtimeVersion passed in is currently only used for showing a warning to the user if the computed fingerprint differs from what the local machine thought it was. 

We need to calculate and override the fingerprint even if the user passed in a runtimeVersion. That being said, from my response to your comment above, we could start trusting and using this field as the override fingerprint rather than re-calculating it.

This is tough.